src/cmd/singularity/cli/actions.go

```diff
@@ -252,6 +252,12 @@ func execStarter(cobraCmd *cobra.Command, image string, args []string, name stri
 	if IpcNamespace {
 		generator.AddOrReplaceLinuxNamespace("ipc", "")
 	}
+	if !UserNamespace {
```

Comment:
Rather than adding an additional check, you can use block `if UserNamespace` line 261 with `else`

Reply:
No, because then it won't behave properly if UserNamespace is set inside the first block.

Reply:
Ah yes !